src/pykeen/triples/triples_factory.py

```diff
@@ -325,24 +393,18 @@ def get_inverse_relation_id(self, relation: int) -> int:
         """Get the inverse relation identifier for the given relation."""
         if not self.create_inverse_triples:
             raise ValueError("Can not get inverse triple, they have not been created.")
-        return self._get_inverse_relation_id(relation)
-
-    @staticmethod
-    def _get_inverse_relation_id(relation_id: Union[int, torch.LongTensor]) -> Union[int, torch.LongTensor]:
-        return relation_id + 1
+        return relation_inverter.get_inverse_id(relation_id=relation)
 
     def _add_inverse_triples_if_necessary(self, mapped_triples: MappedTriples) -> MappedTriples:
         """Add inverse triples if they shall be created."""
         if not self.create_inverse_triples:
             return mapped_triples
 
         logger.info("Creating inverse triples.")
-        h, r, t = mapped_triples.t()
-        r = 2 * r
         return torch.cat(
             [
-                torch.stack([h, r, t], dim=-1),
-                torch.stack([t, self._get_inverse_relation_id(r), h], dim=-1),
+                relation_inverter.map(batch=mapped_triples),
+                relation_inverter.map(batch=mapped_triples, invert=True).flip(1),
```

Comment:
maybe the relation inverter should be an instance variable of the labeling class in case you want to specify a custom one (give in the init a new keyword `relation_inverter: Optional[RelationInverter] = None` that will get the default if not specified

Reply:
I think they should be separate:

The labeling is only considered with converting between string labels and integer IDs; the relation inverter in contrast is purely ID-based and can also be used without any underlying labels.